src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets

```diff
@@ -178,6 +181,19 @@ Copyright (c) .NET Foundation. All rights reserved.
       <PackageReference Condition="'%(PackageReference.Identity)' == 'Microsoft.NETCore.App'">
         <Version Condition="'%(PackageReference.Version)' == ''">$(_TargetFrameworkVersionWithoutV)</Version>
       </PackageReference>
+
+      <!-- Reference the Platforms package in order to supply the RID graph to NuGet.
+           This can be removed once we do https://github.com/dotnet/cli/issues/10528 or
+           https://github.com/NuGet/Home/issues/7351 
+            
+           We only do this here when there are PackageReferences. If the only references 
+           are FrameworkReferences, then we do not need to provide this package to NuGet. -->
+      <PackageReference Include="Microsoft.NETCore.Platforms"
```

Comment:
Is it a problem to move this from eval to build time? Feels like you've hooked it in early enough but maybe VS has other ideas?

Reply:
We are allowed to generate PackageReferences during CollectPackageReference. We're already mutating them in this target, which is why I did it here and figured I'd add a condition to stop doing the whole target when there are no PackageReferences.

Reply:
This could cause issues with package downgrades.  There are some repos that had issues with package downgrades with this package, which they are resolving by doing an update on the `PackageReference`.  So we should probably offer an easier way to control this, for example a property to control the version or whether the reference is added at all.

Reply:
I added an individual opt-out in [950afa9](https://github.com/dotnet/sdk/commit/950afa9164fe71b049b0ab6c1102977f8048aa9c)